python/ray/data/_internal/execution/operators/actor_pool_map_operator.py

```diff
@@ -351,6 +362,23 @@ def _apply_default_remote_args(ray_remote_args: Dict[str, Any]) -> Dict[str, Any
     def get_autoscaling_actor_pools(self) -> List[AutoscalingActorPool]:
         return [self._actor_pool]
 
+    def _manage_actor_restarting_state(self, actor):
+        actor_state = actor._get_local_state()
+        if actor_state != gcs_pb2.ActorTableData.ActorState.ALIVE:
```

Comment:
What happens if `_get_local_state` returns `None`? Looks like we assume that the actor is restarting -- do we need to worry about this edge case?

Reply:
If raylet.pyx:4371 get_local_actor_state() returns None, then actor.py:1561 _get_local_state() can return None.

I think it's defensive to check for None here, given I am not sure about the interface guarantee for get_local_actor_state(). 

Good catch! Let me fix the code.